src/protocol/bigreq.rs

```diff
@@ -32,26 +33,41 @@ pub const X11_EXTENSION_NAME: &str = "BIG-REQUESTS";
 /// send the maximum version of the extension that you need.
 pub const X11_XML_VERSION: (u32, u32) = (0, 0);
 
-/// Opcode for the Enable request
-pub const ENABLE_REQUEST: u8 = 0;
```

Comment:
One could do both: Have a `pub const` and then have an `opcode()` method (that just returns `ENABLE_REQUEST` instead of `0` in this case).

Would that make sense? (I came up with this proposal when I saw the diff for `multithread_test` which changes a `match` into a series of `if`)

Reply:
Yeah I think it can make sense (see my comment at the top of the thread). Want me to do it?

Reply:
Yup, please do.

It just occurred to me: This also makes sense for symmetry with errors and events. They also have this constant.

Reply:
What about using associated constants?

```rust
pub struct FooRequest {
    pub const OPCODE: u8 = 1;
}
```

Reply:
Google finds https://doc.rust-lang.org/edition-guide/rust-2018/trait-system/associated-constants.html which says that associated constants needs Rust 1.20

Reply:
> It just occurred to me: This also makes sense for symmetry with errors and events. They also have this constant.

If we want this, we will have to revert https://github.com/psychon/x11rb/pull/391

Reply:
I've reverted to the previous way of having a global unassociated constant.

If you wanted to do associated constants I think it makes sense to do it for requests, events, and replies all at the same time.